app/controllers/WKRemoteWorkerController.scala

```diff
@@ -76,4 +84,35 @@ class WKRemoteWorkerController @Inject()(jobDAO: JobDAO, jobService: JobService,
       } yield Ok
   }
 
+  def attachVoxelyticsWorkflow(key: String, id: String): Action[String] = Action.async(validateJson[String]) {
```

Comment:
The `workerDAO.findOneByKey` is the permission check, as the key is secret. This way, only workers connected to this webknossos can call this route. Since the worker code is defined by us and will only attach the workflows to its own jobs, this should be enough.

Reply:
Oh, sorry as I do / did not review the worker code, I didn't see that this route is used only by the worker (and also missed that it is not used by the frontend :see_no_evil:)

Thanks for clarifying :pray: